### PR TITLE
Automatic qa

### PR DESCRIPTION
**Pull request by amitaibu**
Blog post + tests ;)

### CODE REVIEW

**Comment by amitaibu** [comment]
Merged.

**Comment by DavidGizra** [comment]
Like.

On Sun, Feb 1, 2015 at 9:00 AM, Amitai Burstein <notifications@github.com>
wrote:

> Blog post + tests ;)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Gizra/Gizra/pull/135
> Commit Summary
>
>    - Automatic QA
>    - Add docs
>    - More
>    - line
>    - Add travis and Behat
>    - Cleanup
>    - more
>    - Install
>    - deubg
>    - d2
>    - fix
>    - d3
>    - d5
>    - d5
>    - More tests
>    - README
>    - fix
>    - d5
>    - Add example
>    - Remove logs
>    - editing
>    - change
>
> File Changes
>
>    - *A* .travis.yml
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-0> (39)
>    - *M* README.md <https://github.com/Gizra/Gizra/pull/135/files#diff-1>
>    (7)
>    - *A* app/_posts/2015-01-30-automatic-qa.md
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-2> (113)
>    - *A* behat/.gitignore
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-3> (4)
>    - *A* behat/README.md
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-4> (9)
>    - *A* behat/behat.local.yml.example
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-5> (4)
>    - *A* behat/behat.local.yml.travis
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-6> (4)
>    - *A* behat/behat.yml
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-7> (13)
>    - *A* behat/composer.json
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-8> (10)
>    - *A* behat/composer.lock
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-9> (1571)
>    - *A* behat/features/blog_post.feature
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-10> (19)
>    - *A* behat/features/bootstrap/FeatureContext.php
>    <https://github.com/Gizra/Gizra/pull/135/files#diff-11> (31)
>
> Patch Links:
>
>    - https://github.com/Gizra/Gizra/pull/135.patch
>    - https://github.com/Gizra/Gizra/pull/135.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Gizra/Gizra/pull/135>.
>



-- 
*Best regards, *
*David Kohav*

File: app/_posts/2015-02-01-automatic-qa.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Here is a known secret - (almost) nobody is doing QA. Since I'm not into arguing about this, I'm willing to soften it a bit to "most companies, don't do proper QA".
+
+The reasons are pretty clear - not enough time and not enough budget. This post isn't going to be about the importance of QA - that point is clear to everybody, but rather give _realistic_ tips and tools that will allow you to start improving the quality of your projects, and actually even save you some time and money.
+
+<!-- more -->
+
+Let's start by agreeing that a developer can never do a proper QA to their own feature. They can't not because they don't care, but because they are normally rushed into their next task.  
+
+Let's also agree that a code review (which for me is even more important than QA, but that's for another post), isn't QA. It may catch bugs - but it's meant to check the entire site or application.
```
Comment (bricel) [inline app/_posts/2015-02-01-automatic-qa.md]: but it's meant to check the entire site or application. =>  but it's not meant to check the entire site or application.

**Comment by amitaibu** [comment]
@bricel @OritGeron Fixed, thanks.

**Pull request closed without merging**